zrml/court/src/benchmarks.rs

```diff
@@ -0,0 +1,205 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use crate::{BalanceOf, Config, CourtPalletApi, MarketIdOf, Pallet};
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_support::traits::{Currency, Get};
+use frame_system::RawOrigin;
+use orml_traits::MultiCurrency;
+use sp_runtime::traits::SaturatedConversion;
+use zeitgeist_primitives::{
+    constants::MinLiquidity,
+    types::{
+        Asset, MarketCreation, MarketEnd, MarketType, MultiHash, OutcomeReport, ScalarPosition,
+    },
+};
+use zrml_market_commons::MarketCommonsPalletApi;
+
+// Get default values for market creation. Also spawns an account with maximum
+// amount of native currency
+fn create_market_common_parameters<T: Config>(
+    permission: MarketCreation,
+) -> Result<
+    (T::AccountId, T::AccountId, MarketEnd<T::BlockNumber>, MultiHash, MarketCreation),
+    &'static str,
+> {
+    let caller: T::AccountId = whitelisted_caller();
+    let _ = T::Currency::deposit_creating(&caller, (u128::MAX).saturated_into());
+    let oracle = caller.clone();
+    let end = <MarketEnd<T::BlockNumber>>::Block((u128::MAX).saturated_into());
+    let mut metadata = [0u8; 50];
+    metadata[0] = 0x15;
+    metadata[1] = 0x30;
+    let creation = permission;
+    Ok((caller, oracle, end, MultiHash::Sha3_384(metadata), creation))
+}
+
+// Create a market based on common parameters
+fn create_market_common<T: Config>(
+    permission: MarketCreation,
+    _options: MarketType,
+) -> Result<(T::AccountId, MarketIdOf<T>), &'static str> {
+    let (caller, _oracle, _end, _metadata, _creation) =
+        create_market_common_parameters::<T>(permission)?;
+
+    //    if let MarketType::Categorical(categories) = options {
+    //        let _ = Call::<T>::create_categorical_market(oracle, end, metadata, creation, categories)
+    //            .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+    //    } else if let MarketType::Scalar(range) = options {
+    //        let _ = Call::<T>::create_scalar_market(oracle, end, metadata, creation, range)
+    //            .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+    //    } else {
+    //        panic!("create_market_common: Unsupported market type: {:?}", options);
+    //    }
+
+    let marketid = T::MarketCommons::latest_market_id()? - 1u32.into();
+    Ok((caller, marketid))
+}
+
+fn create_close_and_report_market<T: Config>(
+    permission: MarketCreation,
+    options: MarketType,
+    _outcome: OutcomeReport,
+) -> Result<(T::AccountId, MarketIdOf<T>), &'static str> {
+    let (caller, marketid) = create_market_common::<T>(permission, options)?;
+    //    let _ = Call::<T>::admin_move_market_to_closed(marketid)
+    //        .dispatch_bypass_filter(T::ApprovalOrigin::successful_origin())?;
+    //    let _ = Call::<T>::report(marketid, outcome)
+    //        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
```

Comment:
@sea212 Just for the sake of benchmarks, commented methods will have to be moved from PM to `MarketCommons`. Can I simply create local functions that mimic them?

Otherwise, what semantically prevents moving other call methods like `report`, `sell_complete_set`, etc?

Reply:
Posting the result of the private discussion:

I think we have to rethink the benchmarks. All this moving around is likely not necessary. For all the changed functions, we have to think again about: What are the logical paths the call can take and on which variables does its execution length depend on? Based on this question we have to rewrite the benchmarks for those functions.
One example is the internal_resolve function. Before it was called within the PM pallet when an admin resolved the market or at on_initialize for each market that was reported or disputed. Now the question arises: How did this functions change and which logical paths can they take now and on which variables does their execution time depend?
The on_initialize function now calls T::Court::on_resolution(now). So we have to understand what this function does in regards to the criteria I mentioned and write appropriate benchmarks for the PM pallet that initialize the storage appropriately for the different cases and then run the benchmark by calling on_intialize. Within on_initialize, we have to distinguish the logical paths and invoke the weight function for the appropriate case.
I just saw that you already created and returned a ResolutionCounters struct, that contains all the relevant parameters that influence the execution time. That is great, now we only have to determine which weight function to call and insert the parameters there (for example, was it a scalar or a categorical market? Does it even make a difference now after the changes or can we unite them?)

c410-f3r:
> I think I already have enough to provide a MVP for these modified benchmarks.
Since it will take some time, can we merge https://github.com/zeitgeistpm/zeitgeist/pull/167 without the finished benchmarks? It would be nice because liquidity mining will use some of the MarketCommons structure.

I think it is fine to merge this and create a issue to adjust those benchmarks and add new weights. The issue must be resolved until the next runtime upgrade, when the new weights will take effect.